### PR TITLE
FIX: Fix for the event buffer growing unbounded in Editor when a cont…

### PR DESCRIPTION
**Pull request by peterhall-unity3d**
…roller is connected and certain options are set.

### Description

Fix for bug https://jira.unity3d.com/browse/UUM-19480 - PS4 Controller causes large CPU lag spikes when connected under certain settings.

The status events received by the Editor are just being kept in the buffer used to pass events into managed code.  As more status events come in, the buffer grows, reallocating and copying the existing events into the new buffer (which is slow as the buffer reaches 50+MB after a few minutes).

### Changes made

When the buffer containing events is passed into managed code, if the settings are causing the buffer to grow (“Background Behaviour” is set to “Ignore Focus” and “Play Mode Input Behaviour” is set to “All Device Input Always Goes To Game View”) then we check the buffer size.  Usually the managed code update would early out and leave events in the buffer at this point, but now when the buffer is larger than 100kB (slightly arbitrary, chosen based on a suggestion that the buffer should not be getting this big) then we will run through the buffer and drop any status events.

### Checklist

Before review:

- [X] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [X] ~~Tests added/changed, if applicable.~~
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [X] ~~Docs for new/changed API's.~~
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ however, it has to be formatted properly to pass verification tests.
 ### Fixed
 
 - Fixed unclosed profiler marker in `InvokeCallbacksSafe_AnyCallbackReturnsTrue` which would lead to eventually broken profiler traces in some cases like using `PlayerInput` (case ISXB-393).
+- Fixed an issue where connecting a gamepad in the editor with certain settings will cause memory and performance to degrade ([case UUM-19480](https://jira.unity3d.com/browse/UUM-19480)).
```
Comment (jimon) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: We use public links which will be `https://issuetracker.unity3d.com/product/unity/issues/guid/UUM-19480` in this case

**Review by jimon — APPROVED** [review]

**Comment by jimon** [comment]
Looks good if it works, thanks for the help!

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (peterhall-unity3d) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Thank you!
(fixed just below in commit a1bf6e1)

**Comment by jimon** [comment]
@peterhall-unity3d feel free to rebase on the latest develop to resolve CI issues

**Comment by peterhall-unity3d** [comment]
> @peterhall-unity3d feel free to rebase on the latest develop to resolve CI issues

Awesome, thanks for letting me know it was fixed! 😄

**Pull request merged**